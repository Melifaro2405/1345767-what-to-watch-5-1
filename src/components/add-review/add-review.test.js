import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import AddReview from './add-review';

jest.mock(`../user-block/user-block`, () => `UserBlock`);
jest.mock(`../form-review/form-review`, () => `FormReview`);

const film = {
  id: 1,
  preview: {
    src: `test`,
    title: `test`
  },
  moreInfo: {
    backgroundSrc: `test`,
    backgroundColor: `test`,
    posterSrc: `test`,
    genre: `test`,
    releaseDate: 2014,
    playPreviewSrc: `test`,
    playVideoSrc: `test`,
    isAddToMyList: false
  },
  overview: {
    description: `test`,
    rating: 8.9,
    ratingCount: 240,
    director: `test`,
    actorsList: [`test`]
  },
  details: {
    runtime: 99
  }
};

it(`Should AddReview render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <AddReview film={film} />
        </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
