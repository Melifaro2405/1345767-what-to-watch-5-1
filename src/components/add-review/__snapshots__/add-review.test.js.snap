// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddReview render correctly 1`] = `
<section
  className="movie-card movie-card--full"
>
  <div
    className="movie-card__header"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="test"
        src="test"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <nav
        className="breadcrumbs"
      >
        <ul
          className="breadcrumbs__list"
        >
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
              href="/films/1"
              onClick={[Function]}
            >
              test
            </a>
          </li>
          <li
            className="breadcrumbs__item"
          >
            <a
              className="breadcrumbs__link"
            >
              Add review
            </a>
          </li>
        </ul>
      </nav>
      <UserBlock />
    </header>
    <div
      className="movie-card__poster movie-card__poster--small"
    >
      <img
        alt="test"
        height="327"
        src="test"
        width="218"
      />
    </div>
  </div>
  <div
    className="add-review"
  >
    <FormReview
      changeIsError={[Function]}
      changeIsLoading={[Function]}
      filmID={1}
      isActive={false}
      isError={false}
      isLoading={false}
      onChangeRating={[Function]}
      onChangeText={[Function]}
      rating=""
      text=""
    />
  </div>
</section>
`;
